### PR TITLE
Explain #5713: KDoc line breaks are whitespaces

### PR DESCRIPTION
**Pull request by TWiStErRob**
* Closes https://github.com/detekt/detekt/issues/5713 as won't fix for now.

Implementing my suggestions at
 * https://github.com/detekt/detekt/pull/6370#pullrequestreview-1572622285
 * https://github.com/detekt/detekt/pull/6370#issuecomment-1804821373

### CODE REVIEW

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/TrailingWhitespace.kt [hunk 1]
```diff
@@ -14,6 +14,12 @@ import org.jetbrains.kotlin.psi.KtFile
 
 /**
  * This rule reports lines that end with a whitespace.
+ *
+ * Note: in KDoc comments we use Markdown, so two spaces at the end of lines should be allowed.
+ * However, JetBrains haven't implemented this in their flavour of "standard" Markdown yet (
+ * [in Dokka](https://github.com/Kotlin/dokka/issues/2823),
+ * nor [in KTIJ](https://youtrack.jetbrains.com/issue/KTIJ-6702/KDoc-Dokka-allow-for-newlines-line-breaks-inside-paragraphs)
+ * ), which means Markdown line-breaks in KDoc are really only trailing whitespace for now.
```
Comment (TWiStErRob) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/TrailingWhitespace.kt:22]: Preview: https://detekt-rf4a20o8f-detekt.vercel.app/docs/next/rules/style#trailingwhitespace

image

Comment (atulgpt) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/TrailingWhitespace.kt:22]: Should we also point to the JetBrains issue id?

Comment (atulgpt) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/TrailingWhitespace.kt:22]:
```suggestion
 * However, JetBrains haven't implemented this in their flavour of "standard" Markdown yet
 * ([in Dokka](https://github.com/Kotlin/dokka/issues/2823),
 * nor [in KTIJ](https://youtrack.jetbrains.com/issue/KTIJ-6702/KDoc-Dokka-allow-for-newlines-line-breaks-inside-paragraphs)),
 * which means Markdown line-breaks in KDoc are really only trailing whitespace for now.
```

Are spaces after `(` and before `)` neccessary?

Comment (TWiStErRob) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/TrailingWhitespace.kt:22]: Wanted to keep the links on separate lines, but your suggestion is rendered better and doesn't complicate the code to much

**Review by BraisGabin — APPROVED** [review]

**Pull request merged**